packages/react/menu/src/Menu.tsx

```diff
@@ -74,13 +73,26 @@ type MenuOwnProps = {
 const Menu: React.FC<MenuOwnProps> = (props) => {
   const { open = false, children, onOpenChange } = props;
   const [content, setContent] = React.useState<MenuContentElement | null>(null);
+  const [isKeyboardUser, setIsKeyboardUser] = React.useState(false);
```

Comment:
Like you said, probably doesn't need to be state as it doesn't affect rendering.

Reply:
Not a fan of the name `isKeyboardUser` as well, it's a bit misleading to me because for example I use both keyboard and mouse. To me it's more the last intent used I guess.

Reply:
> Like you said, probably doesn't need to be state as it doesn't affect rendering.

Turns out i can't do this because when `context.open` state changes at submenu level that state change obviously won't trigger a re-render of the parent context so it wouldn't have the updated reference. Also, when a `menu` is open you can switch from being a keyboard user to a mouse user but the ref change wouldn't update the context consumers (obvs) so this happens:

![CleanShot 2021-06-10 at 11 50 22](https://user-images.githubusercontent.com/175330/121512849-152d1500-c9e2-11eb-8758-c7d3a82c59ee.gif)

`onItemLeave` is firing and because it thinks i'm a keyboard user still, it's focusing the first item. So yeah, needs to be state.

> it's a bit misleading to me because for example I use both keyboard and mouse

Would `isUsingKeyboard` make more sense?

Reply:
I'm not sure I'm following. If the ref itself was being passed down, then `.current` on it would always be up to date no?

`isUsingKeyboard` sounds good to me 👍

Reply:
A `ref` change doesn't won't trigger provider consumers to re-render so the consumers will still have the previous ref from the previous render closure.

Reply:
Oh wait, the ref would be fine if i passed the whole ref lol, it's cos I was passing `isKeyboardUser={isKeyboarRef.curren}` ... my bad 😅  too much COVID in my veins

Reply:
Those were my thoughts :0